api/versions.go

```diff
@@ -326,12 +329,38 @@ func (self *version2_0) HandleRequest(requestType string, request []string, m ma
 		query := info.ContainerInfoRequest{
 			NumStats: sr.Count,
 		}
-		cont, err := m.GetContainerInfo(name, &query)
-		if err != nil {
-			return fmt.Errorf("failed to get container %q: %v", name, err)
+		switch sr.IdType {
+		case typeName:
+			cont, err := m.GetContainerInfo(name, &query)
+			if err != nil {
+				return fmt.Errorf("failed to get container %q: %v", name, err)
+			}
+			contStats := convertStats(cont)
+			return writeResult(contStats, w)
+		case typeDocker:
+			contStats := make(map[string][]v2.ContainerStats, 0)
+			if name == "/" {
+				// special case: get all docker containers.
+				// TODO(rjnagal): require recursive=true to be set?
+				containers, err := m.AllDockerContainers(&query)
+				if err != nil {
+					return fmt.Errorf("failed to get all docker containers: %v", err)
+				}
+				for name, cont := range containers {
+					contStats[name] = convertStats(&cont)
+				}
+			} else {
+				name = path.Base(name)
```

Comment:
lets to strings.TrimPrefix(name, "/") so that we don't ignore if the users gives us: /this/is/bad/<docker ID>